packages/dd-trace/src/appsec/waf/index.js

```diff
@@ -99,8 +108,18 @@ function run (data, req, raspRule) {
   }
 
   const wafContext = waf.wafManager.getWAFContext(req)
+  const result = wafContext.run(data, raspRule)
+
+  if (result?.keep) {
+    if (limiter.isAllowed()) {
+      const rootSpan = web.root(req)
+      keepTrace(rootSpan, ASM)
+    } else {
+      updateRateLimitedMetric(req)
+    }
+  }
```

Comment:
i don't understand why this has to live in this file, and not be in the reporter like it was before ? the waf reports its output and state to the reporter, which then takes care of sending it to the backend

Reply:
the rate limiter and metrics used to depend on the presence of event=true. Now it depends on the presence of keep=true. That doesn't change anything to the underlying organization of the code no ? just change the field that's used :shrug:

Reply:
alright we zoomed, and i got convinced enough to keep it this way :+1: